frontend/packages/dev-console/src/components/import/image-search/ImageStream.tsx

```diff
@@ -2,79 +2,67 @@ import * as React from 'react';
 import * as _ from 'lodash';
 import { Alert } from '@patternfly/react-core';
 import { useFormikContext, FormikValues } from 'formik';
-import { k8sGet } from '@console/internal/module/k8s';
-import { ImageStreamTagModel, RoleBindingModel } from '@console/internal/models';
-import { useAccessReview } from '@console/internal/components/utils';
-import { DropdownField, ResourceDropdownField, CheckboxField } from '@console/shared';
-import { ImageStreamProps } from '../import-types';
+import { CheckboxField } from '@console/shared';
 import {
-  getSuggestedName,
-  getPorts,
-  makePortName,
   RegistryType,
-  getProjectResource,
   BuilderImagesNamespace,
-  getImageStreamByNamespace,
   getImageStreamTags,
 } from '../../../utils/imagestream-utils';
+import { ImageStreamState, ImageStreamAction, ImageStreamActions } from '../import-types';
+import { ImageStreamContext } from './ImageStreamContext';
+import ImageStreamNsDropdown from './ImageStreamNsDropdown';
+import ImageStreamDropdown from './ImageStreamDropdown';
+import ImageStreamTagDropdown from './ImageStreamTagDropdown';
+
 import './ImageStream.scss';
 
-const ImageStream: React.FC<ImageStreamProps> = ({ imageStreams }) => {
-  const resources = getProjectResource();
-  let imageStreamList = {};
-  let imageStreamTagList = {};
+export const initialState: ImageStreamState = {
+  hasAccessToPullImage: true,
+  loading: false,
+  accessLoading: false,
+  hasCreateAccess: false,
+  selectedImageStream: {},
+};
+
+export const ImageStreamReducer = (state: ImageStreamState, action: ImageStreamAction) => {
+  const { value } = action;
+  switch (action.type) {
+    case ImageStreamActions.setHasAccessToPullImage:
+      return { ...state, hasAccessToPullImage: value };
+    case ImageStreamActions.setLoading:
+      return { ...state, loading: value };
+    case ImageStreamActions.setAccessLoading:
+      return { ...state, accessLoading: value };
+    case ImageStreamActions.setHasCreateAccess:
+      return { ...state, hasCreateAccess: value };
+    case ImageStreamActions.setSelectedImageStream:
+      return { ...state, selectedImageStream: value };
+    default:
+      return initialState;
```

Comment:
This is a closed React reducer, perhaps throw an error here so there is no accidental mis-usage. 

Got the idea from here: https://reactjs.org/docs/hooks-reference.html#usereducer